### PR TITLE
AUTH-5770 Add support for OIDC SaaS Apps to Access Applications

### PR DESCRIPTION
**Pull request by ajholland**
## Description

https://developers.cloudflare.com/api/operations/access-applications-list-access-applications 

## Has your change been tested?

added unit tests

## Types of changes

What sort of change does your code introduce/modify?

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x ] My code follows the code style of this project.
- [ x] My change requires a change to the documentation.
- [ x] I have updated the documentation accordingly.
- [ x] I have added tests to cover my changes.
- [ x] All new and existing tests passed.
- [ x] This change is using publicly documented in [cloudflare/api-schemas](https://github.com/cloudflare/api-schemas) 
      and relies on stable APIs.

[1]: https://help.github.com/articles/closing-issues-using-keywords/

### CODE REVIEW

**Review by jacobbednarz — APPROVED** [review]
nice one 👏 once we have docs, this is good to land.

**Comment by ajholland** [comment]
@jacobbednarz updated docs link, if you could merge this when available, thanks!

**Comment by jacobbednarz** [comment]
legend! thanks

**Pull request merged**
